src/java/org/apache/cassandra/cql3/restrictions/SingleColumnRestriction.java

```diff
@@ -509,6 +519,21 @@ protected boolean isSupportedBy(Index index)
             return supported;
         }
 
+        @Override
+        public boolean needsFiltering(Index.Group indexGroup)
+        {
+            // multiple contains might require filtering on some indexes, since that is equivalent to a disjunction (or)
+            boolean hasMultipleContains = numberOfValues() + numberOfKeys() + numberOfEntries() > 1;
```

Comment:
Despite operator precedence playing nice here I would rewrite for clarity to:
`boolean hasMultipleContains = (numberOfValues() + numberOfKeys() + numberOfEntries()) > 1;`

Reply:
Done.